### PR TITLE
For #2329 - Adds a parameter to the cancel probe to show the type 🌮

### PR DESCRIPTION
**Pull request by boek**

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/menu/context/WebContextMenu.java [hunk 1]
```diff
@@ -66,9 +66,19 @@ public static void show(final @NonNull Context context, final @NonNull IWebView.
         builder.setOnCancelListener(new DialogInterface.OnCancelListener() {
             @Override
             public void onCancel(DialogInterface dialog) {
+                // What type of element was long-pressed
+                String type;
```
Comment (ekager) [inline app/src/main/java/org/mozilla/focus/menu/context/WebContextMenu.java]: After using Kotlin this seems archaic to me 😅

**Comment by pocmo** [comment]
Looks like taskcluster died. I'll retrigger.

**Comment by boek** [comment]
I changed it from a parameter to a value to match all of the other pings of the same type: https://github.com/mozilla-mobile/focus-android/wiki/Telemetry

**Comment by ekager** [comment]
Hm I think usually that's a nonchanging value just to differentiate between events with the same category etc but I see what you mean about the similar events. https://github.com/mozilla-mobile/focus-android/wiki/Telemetry#context-menu
@pocmo should this be an extra or a value?

**Comment by boek** [comment]
@ekager That seems to be how it's being used here?

image

I was planning on splitting up the cancel into multiple lines like `share`/`copy`. How does that sound?

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt [hunk 2]
```diff
@@ -150,6 +151,17 @@ object TelemetryWrapper {
         val AVERAGE = "average"
     }
 
+
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: \[ktlint\] WARNING: \[no\-consecutive\-blank\-lines\] Needless blank line\(s\) in column 1

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt [hunk 3]
```diff
@@ -150,6 +151,17 @@ object TelemetryWrapper {
...
+    enum class BrowserContextMenuValue {
+        Link, Image, ImageWithLink;
+
+        override fun toString(): String = when(this) {
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: \[ktlint\] WARNING: \[keyword\-spacing\] Missing spacing after "when" in column 47

File: app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt [hunk 4]
```diff
@@ -150,6 +151,17 @@ object TelemetryWrapper {
...
+            Link -> Value.LINK
+            Image -> Value.IMAGE
+            ImageWithLink -> Value.IMAGE_WITH_LINK
+       }
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt]: \[ktlint\] WARNING: \[indent\] Unexpected indentation \(7\) \(it should be multiple of 4\) in column 1

**Comment by pocmo** [comment]
Looks good to me. :)

**Pull request merged**